src/Microsoft.CodeQuality.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.cs

```diff
@@ -0,0 +1,357 @@
+        {
+            if (!TryGetInt32Constant(binaryOperation.LeftOperand, out var constant))
+            {
+                return false;
+            }
+
+            if (constant == 0 &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.Equals &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.NotEquals &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.LessThan &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.GreaterThanOrEqual)
+            {
+                return false;
+            }
+            else if (constant == 1 &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.LessThanOrEqual &&
+                binaryOperation.OperatorKind != BinaryOperatorKind.GreaterThan)
+            {
+                return false;
+            }
+            else if (constant > 1)
+            {
+                return false;
+            }
+
+            return IsCountMethodInvocation(binaryOperation.RightOperand, enumerableType);
+        }
+
+        /// <summary>
+        /// Tries the get an <see cref="int"/> constant from the <paramref name="operation"/>.
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        /// <param name="constant">The constant the <paramref name="operation"/> represents, if succeeded, or zero if <paramref name="operation"/> is not a constant.</param>
+        /// <returns><see langword="true" /> <paramref name="operation"/> is a constant, <see langword="false" /> otherwise.</returns>
+        public static bool TryGetInt32Constant(IOperation operation, out int constant)
+        {
+            constant = default;
+
+            if (operation.Type?.SpecialType != SpecialType.System_Int32)
+            {
+                // TODO: Should we be checking for other integer types?
+                return false;
+            }
+
+            var comparandValueOpt = operation.ConstantValue;
+
+            if (!comparandValueOpt.HasValue)
+            {
+                return false;
+            }
+
+            constant = (int)comparandValueOpt.Value;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Checks the is a <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource})" />
+        /// or <see cref="System.Linq.Enumerable.Count{TSource}(System.Collections.Generic.IEnumerable{TSource}, Func{TSource, bool})" /> invocation.
+        /// </summary>
+        /// <param name="operation">The operation.</param>
+        /// <param name="enumerableType">Type of the enumerable.</param>
+        /// <returns><see langword="true" /> if XXXX, <see langword="false" /> otherwise.</returns>
+        private static bool IsCountMethodInvocation(IOperation operation, INamedTypeSymbol enumerableType)
+        {
+            if (operation.Kind != global::Microsoft.CodeAnalysis.OperationKind.Invocation ||
```

Comment:
Any reason to use fully qualified names instead of just having a `using Microsoft.CodeAnalysis` and `using Microsoft.CodeAnalysis.Operations`?